flink-cdc-connect/flink-cdc-pipeline-connectors/flink-cdc-pipeline-connector-kafka/src/main/java/org/apache/flink/cdc/connectors/kafka/json/debezium/DebeziumJsonRowDataSerializationSchema.java

```diff
@@ -0,0 +1,148 @@
+ */
+public class DebeziumJsonRowDataSerializationSchema implements SerializationSchema<RowData> {
+    private static final long serialVersionUID = 1L;
+
+    /** RowType to generate the runtime converter. */
+    private final RowType rowType;
+
+    /** The converter that converts internal data formats to JsonNode. */
+    private final RowDataToJsonConverters.RowDataToJsonConverter runtimeConverter;
+
+    /** Object mapper that is used to create output JSON objects. */
+    private transient ObjectMapper mapper;
+
+    /** Reusable object node. */
+    private transient ObjectNode node;
+
+    /** Timestamp format specification which is used to parse timestamp. */
+    private final TimestampFormat timestampFormat;
+
+    /** The handling mode when serializing null keys for map data. */
+    private final JsonFormatOptions.MapNullKeyMode mapNullKeyMode;
+
+    /** The string literal when handling mode for map null key LITERAL. */
+    private final String mapNullKeyLiteral;
+
+    /** Flag indicating whether to serialize all decimals as plain numbers. */
+    private final boolean encodeDecimalAsPlainNumber;
+
+    private final boolean includeSchemaInfo;
+
+    public DebeziumJsonRowDataSerializationSchema(
+            RowType rowType,
+            TimestampFormat timestampFormat,
+            JsonFormatOptions.MapNullKeyMode mapNullKeyMode,
+            String mapNullKeyLiteral,
+            boolean encodeDecimalAsPlainNumber,
+            boolean includeSchemaInfo) {
+        this.rowType = rowType;
+        this.timestampFormat = timestampFormat;
+        this.mapNullKeyMode = mapNullKeyMode;
+        this.mapNullKeyLiteral = mapNullKeyLiteral;
+        this.encodeDecimalAsPlainNumber = encodeDecimalAsPlainNumber;
+        this.runtimeConverter =
+                new RowDataToJsonConverters(timestampFormat, mapNullKeyMode, mapNullKeyLiteral)
+                        .createConverter(rowType);
+        this.includeSchemaInfo = includeSchemaInfo;
+    }
+
+    @Override
+    public void open(InitializationContext context) throws Exception {
+        mapper =
+                JacksonMapperFactory.createObjectMapper()
+                        .configure(
+                                JsonGenerator.Feature.WRITE_BIGDECIMAL_AS_PLAIN,
+                                encodeDecimalAsPlainNumber);
+    }
+
+    @Override
+    public byte[] serialize(RowData row) {
+        if (node == null) {
+            node = mapper.createObjectNode();
+        }
+
+        try {
+            runtimeConverter.convert(mapper, node, row);
+            if (includeSchemaInfo) {
+                // schema is a nested json string, asText() can return a pure string without other
+                // escape characters such as "\"
+                String schemaValue = node.get("schema").asText();
+                JsonNode schemaNode = mapper.readTree(schemaValue);
+                node.set("schema", schemaNode);
```

Comment:
Because the schema is passed to the downstream as a string, and there is a nested json in the schema, if the json string is put into jsonNode, there will be [\"]. The JsonNode.asText() method can solve this problem well.